frontend/src/lib/utils.tsx

```diff
@@ -1579,6 +1579,19 @@ export function isNumeric(x: any): boolean {
     return !isNaN(Number(x)) && !isNaN(parseFloat(x))
 }
 
+export function isNull<T>(arg: T | null): arg is null {
+    return arg === null
+}
+export function isNotNull<T>(arg: T): arg is Exclude<T, null> {
+    return arg !== null
+}
+export function isNil<T>(arg: T | null | undefined): arg is null | undefined {
+    return arg === null || arg === undefined
+}
+export function isNotNil<T>(arg: T): arg is Exclude<T, null | undefined> {
+    return arg !== null && arg !== undefined
+}
+
```

Comment:
It's useful to have the as typeguards, because it allows you to do

```typescript
const myList = [1, 2, null]; // type is (number | null)[]

const myFilteredList = list.filter(isNotNil) // type is number[]

const squaredList = list.map(x => x * x) // not a type error!
```

However if we don't like having the functions here I could just pull them in from lodash. I could also keep them here but add a code example explaining why they are useful.

Reply:
I sometimes get the impression that TypeScript should be smart enough to restrict the type if you filter on it, but then other times it does surprise me. I didn't think of this aspect of this, apologies. My comment came from a place, where I've seen a lot of well meaning, yet unused global utility functions over the years. Overall I don't care too strongly here.. other than "introducing lodash" is probably another big project with dubious gains 😅

Reply:
We already have lodash 😅  well a couple of one-off packages like lodash.memoize

Reply:
I went the route of just keeping the one I use, and adding a comment explaining why it's useful